modules/katello_lifecycle_environment.py

```diff
@@ -0,0 +1,198 @@
+options:
+    server_url:
+        description:
+            - URL of Foreman server
+        required: true
+    username:
+        description:
+            - Username on Foreman server
+        required: true
+    password:
+        description:
+            - Password for user accessing Foreman server
+        required: true
+    verify_ssl:
+        description:
+            - Verify SSL of the Foreman server
+        default: false
+    name:
+        description:
+            - Name of the lifecycle environment
+        required: true
+    label:
+        description:
+            - Label of the lifecycle environment. This field cannot be updated.
+    description:
+        description:
+            - Description of the lifecycle environment
+    organization:
+        description:
+            - Organization name that the lifecycle environment is in
+        required: true
+    prior:
+        description:
+            - Name of the parent lifecycle environment
+    state:
+        description:
+            - Whether the lifecycle environment should be present or absent on the server
+        required: true
+        choices:
+            - absent
+            - present
+'''
+
+EXAMPLES = '''
+- name: "Add a production lifecycle environment"
+  katello_lifecycle_environment:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://katello.example.com"
+    verify_ssl: false
+    name: "Production"
+    label: "production"
+    organization: "Default Organization"
+    prior: "Library"
+    description: "The production environment"
+    state: "present"
+'''
+
+RETURN = '''# '''
+
+try:
+    from nailgun.entities import (
+        LifecycleEnvironment
+    )
+    HAS_NAILGUN_PACKAGE = True
+except:
+    HAS_NAILGUN_PACKAGE = False
+
+from ansible.module_utils.ansible_nailgun_cement import (
+    create_server,
+    ping_server,
+    find_organization,
+    find_lifecycle_environment,
+    update_fields,
+)
+
+
+def validate_params(module, state, label=None, description=None, prior=None):
+    message = ""
+    if state != 'present':
+        if label is not None:
+            message += "Label cannot be specified if state is present. "
+        if description is not None:
+            message += "Description cannot be specified if state is present. "
+        if prior is not None:
+            message += "Prior cannot be specified if state is present. "
+    if len(message) > 0:
+        module.fail_json(msg=message)
+        return False
+
+
+def find_prior(module, prior, organization):
+    if prior is not None:
+        return find_lifecycle_environment(module, prior, organization)
+
+
+def lifecycle_environment(module, name, organization, state, label=None, description=None, prior=None):
+    changed = False
+    organization = find_organization(module, organization)
+    prior = find_prior(module, prior, organization)
+    current_le = find_lifecycle_environment(module, name, organization, failsafe=True)
```

Comment:
Seems like it's actually a [hyphenated word](https://www.merriam-webster.com/dictionary/failsafe). Generally, we'd use `fail_safe` if the word/phrase was "fail safe", but since it's "fail-safe," it's kind of an edge case and could go either way. I'm willing to change it if you'd like.